inst/rmd/h/navigation-1.1/tabsets.js

```diff
@@ -75,58 +73,155 @@ window.buildTabsets = function(tocID) {
-      heading.remove();
+      var tabHeading = tab.find('h' + tabLevel + ':first');
+      var tabHeadingText = tabHeading.html();
+      tabHeading.remove();
 
       // build and append the tab list item
-      var a = $('<a role="tab" data-toggle="tab">' + headingText + '</a>');
-      a.attr('href', '#' + id);
-      a.attr('aria-controls', id);
       var li = $('<li role="presentation"></li>');
-      li.append(a);
       tabList.append(li);
 
-      // set it's attributes
-      tab.attr('role', 'tabpanel');
-      tab.addClass('tab-pane');
-      tab.addClass('tabbed-pane');
-      if (fade)
-        tab.addClass('fade');
+      // see if this is marked as the active tab, or set the first tab by
+      // default
+      if (tab.hasClass('active')) {
+        activeTabIndex = i;
+      }
+
+      // check if this is a dropdown tab and process it accordingly
+      var dropdown = tab.hasClass("tabset-dropdown");
+      if (dropdown) {
+        li.addClass('dropdown');
+
+        // build and append the dropdown toggle
+        var a = $('<a class="dropdown-toggle" data-toggle="dropdown" href="#">'
+          + tabHeadingText
+          + ' <span class="caret"></span></a>');
+        li.append(a);
+
+        // build and append the dropdown menu
+        var dropdownMenu = $('<ul class="dropdown-menu"></ul>');
+        li.append(dropdownMenu)
+
+        // determine the heading level of the dropdown tabs
+        var match = tab.attr('class').match(/level(\d) /);
+        if (match === null)
+          return;
+        var dropdownLevel = Number(match[1]);
+        var dropdownTabLevel = dropdownLevel + 1;
+
+        // find all subheadings immediately below
+        var dropdownTabs = tab.find("div.section.level" + dropdownTabLevel);
+        if (!dropdownTabs.length)
+          return;
+
+        dropdownTabs.each(function(j) {
+
+          // get the dropdown tab div
+          var dropdownTab = $(dropdownTabs[j]);
+
+          // get the id then sanitize it for use with bootstrap tabs
+          var dropdownId = dropdownTab.attr('id');
+
+          // remove any table of contents entries associated with
+          // this ID (since we'll be removing the heading element)
+          $("div#" + tocID + " li a[href='#" + dropdownId + "']")
+            .parent().remove();
+
+          // sanitize the id for use with bootstrap tabs
+          dropdownId = dropdownId
+            .replace(/[.\/?&!#<>]/g, '')
+            .replace(/\s/g, '_');
+          dropdownTab.attr('id', dropdownId);
+
+          // get the heading element within it, grab it's text, then remove it
+          var dropdownHeading = dropdownTab
+            .find('h' + dropdownTabLevel + ':first');
+          var dropdownHeadingText = dropdownHeading.html();
+          dropdownHeading.remove();
+
+          // build and append the dropdown tab list item
+          var dropdownLi = $('<li role="presentation"></li>');
+          dropdownMenu.append(dropdownLi);
```

Comment:
Could we make a JS function as this is the same process as for other tabs above but with different parameter ? Refactoring would help later maintaining